### ISSUE TITLE
Handle error in user.go functions

### ISSUE THREAD

**Issue by bhuvansingla** [bhuvansingla/iitk-coin#9]
The following functions in `account/user.go` that access the database have unhandled errors.
https://github.com/bhuvansingla/iitk-coin/blob/4420b8c8da9b3b2874280f96992ae9e159b82dc1/account/user.go#L33
https://github.com/bhuvansingla/iitk-coin/blob/4420b8c8da9b3b2874280f96992ae9e159b82dc1/account/user.go#L40
https://github.com/bhuvansingla/iitk-coin/blob/4420b8c8da9b3b2874280f96992ae9e159b82dc1/account/user.go#L47

Modify these to return the error. Take reference from other existing functions. Also, modify their usages accordingly.

**Comment by erikvega** [comment]
cool if i pick up one more?

**Comment by 1-Harshit** [comment]
Sure, go ahead!

**Comment by erikvega** [comment]
so something like

```
func GetAccountRoleByRollno(rollno string) (Role, error) {
	row := database.DB.QueryRow("SELECT role FROM ACCOUNT WHERE rollno=?", rollno)
	var role Role
	err := row.Scan(&role)
	if err != nil{
		return NormalUser, err
	}
	return role, nil
}
```

where `NormalUser` is 0?

**Comment by erikvega** [comment]
sweet! also probably don't need to throw an error for `UserExists` func.
https://github.com/bhuvansingla/iitk-coin/blob/2c1875ef4602bb3a54f9fc6013106657c59cd587/account/transfer.go#L19-L21
the way it's being used, we already throw an error if a user isn't known to exist. throwing another error would be redundant and mess up the flow of how it's being used.

**Comment by bhuvansingla** [comment]
You are right that we are throwing an error, but that is an HTTP status error. We can still get an error in the line:
https://github.com/bhuvansingla/iitk-coin/blob/4420b8c8da9b3b2874280f96992ae9e159b82dc1/account/user.go#L36 that goes unhandled.

The function will need to be redefined like you did with the last one, and after that the usage will need to be modified accordingly.

**Comment by erikvega** [comment]
okay! i'll mess around with it then

**Comment by bhuvansingla** [comment]
Closing this now.
